CHANGELOG.md

```diff
@@ -1,5 +1,78 @@
 # Changelog
 
+## 1.10.0
+
+- ### Features
+
+  - **`wrangler dev` now requires that you specify "preview" versions of KV namespaces - [EverlastingBugstopper], [issue/1032] [pull/1357] [pull/1360]**
+
+    We're preparing to make `wrangler dev` run entirely on Cloudflare's edge, giving it access to functionality that's currently missing in the preview environment, and more accurately reproducing what happens when your Worker is deployed to production. This means that, among other things, your Workers can access the Cache API and your production KV Namespaces.
+
+    In order to prevent you from accidentally stomping on production data in Workers KV or the Cache API, we're introducing the concept of explicit *preview namespaces*. When running `wrangler dev`, if you're using Workers KV, you'll need to specify a specific KV Namespace to use when previewing the Worker.
+
+    Specifically, this change:
+    * Adds a `preview_id` field to items in `kv_namespaces` in `wrangler.toml` that _must_ be provided in order to preview a worker that has kv namespaces.
+    * also adds `--preview` to kv commands in order to interact with them instead of production namespaces.
+
+    If you define a KV Namespace in your `wrangler.toml` but don't specify a `preview_id`, and then try to run `wrangler dev`, you'll see the following:
+    ```console
+    $ wrangler preview
+    💁  JavaScript project found. Skipping unnecessary build!
+    Error: In order to preview a worker with KV namespaces, you must designate a preview_id for each KV namespace you'd like to preview.
+    ```
+
+    More details can be found in the documentation.
```

Comment:
i still need to write the docs for this - aaannndd we should link to them

Reply:
Haha, yep, that's mostly why I left this to remind both of us!

Reply:
https://github.com/cloudflare/workers-docs/pull/904